subprojects/groovy-templates/src/main/groovy/groovy/text/GStringTemplateEngine.java

```diff
@@ -178,7 +179,8 @@ private static class GStringTemplate implements Template {
 
             templateExpressions.append("}}");
 
-            final GroovyClassLoader loader = parentLoader instanceof GroovyClassLoader?(GroovyClassLoader)parentLoader:(
+            // Use a new class loader by default for each class so each class can be independently garbage collected
+            final GroovyClassLoader loader = reuseClassLoader && parentLoader instanceof GroovyClassLoader?(GroovyClassLoader)parentLoader:(
```

Comment:
Hello, I noticed that this PR is causing a regression. 

Assuming we don't use the reuseClassLoader system property, the actual behavior is that a new class loader will be created using the following constructor:

`GroovyClassLoader(ClassLoader loader)` which sets the CompilerConfiguration to null

Before this change the created class loader will be created depending on `parentLoader` type with

`GroovyClassLoader(GroovyClassLoader parent)` which sets the CompilerConfiguration to parent's config if `parentLoader instanceof GroovyClassLoader`
OR
`GroovyClassLoader(ClassLoader loader)` which sets the CompilerConfiguration to null otherwise

WDYT ?

Thanks